package/com.unity.formats.usd/Runtime/Scripts/IO/Geometry/MeshImporter.cs

```diff
@@ -284,6 +284,9 @@ public class GeometrySubsets {
       Material mat = renderer.sharedMaterial;
       bool changeHandedness = options.changeHandedness == BasisTransformation.SlowAndSafe ||
                               options.changeHandedness == BasisTransformation.SlowAndSafeAsFBX;
+      // Check if the handedness needs to change.
+      changeHandedness = changeHandedness && (usdMesh.orientation != Orientation.LeftHanded) ||
+                !changeHandedness && (usdMesh.orientation == Orientation.LeftHanded);
```

Comment:
When mixing && and ||, you should add parentheses to be sure the logic works the way you expect and to make it clear to future readers